broker/src/test/java/io/camunda/zeebe/broker/engine/RecordVersionTest.java

```diff
@@ -116,7 +116,7 @@ private void deployProcess(final BpmnModelInstance process) {
         .command()
         .put("resources", List.of(Map.of("resourceName", "process.bpmn", "resource", resource)))
         .done()
-        .send();
+        .sendAndAwait();
```

Comment:
Without this change the tests in this class would fail because (apparently) processes were started before they were deployed. To be honest, I don't fully understand why so I wan't to draw attention to this and see if someone thinks this is concerning.

Reply:
This looks like a bug. Either it is re-writing a buffer while the first command is still being written. Or the requests are handled by CommanApi in different order than expected. Could you please verify this?

Reply:
When deploying, waiting 100ms and then creating instances the requests look like this:

```
Handling request 0 CREATE DEPLOYMENT
Handling request 1 CREATE PROCESS_INSTANCE_CREATION
```

When deploying and then creating instances without waiting we get this:
```
Handling request 0 CREATE PROCESS_INSTANCE_CREATION
Handling request 1 CREATE PROCESS_INSTANCE_CREATION
```

So I think you are right that we are probably reusing buffers where we shouldn't. I'll look into this more.

Reply:
Adding a bit more information, the decoder for request 0 and 1 is using the same buffer for reading the command: 

```
Handling request 0 CREATE PROCESS_INSTANCE_CREATION buffer: UnsafeBuffer{addressOffset=16, capacity=47, byteArray=[B@77725fe0, byteBuffer=null}
Handling request 1 CREATE PROCESS_INSTANCE_CREATION buffer: UnsafeBuffer{addressOffset=16, capacity=47, byteArray=[B@77725fe0, byteBuffer=null}
```

which should not happen because the decoder is re-wrapped over the buffer for every request.

Reply:
I'm pretty sure this is a design flaw in this PR. Based on the implementation of the query api I assumed that request handlers could be actors that schedule handling of individual requests. This assumption does not hold:
`AtomixServerTransport.handleAtomixRequest` schedules a call to `RequestHandler.onRequest` with a reused buffer of request bytes. If the `onRequest` implementation then does nothing but schedule a new job (let's call it `RequestJob`), another call to `AtomixServerTransport.handleAtomixRequest` modifies the reused buffer before the `RequestJob` ever ran. When it eventually runs, it has the wrong buffer. That's why we are seeing the same request data being handled twice.
As far as I understand the query api already has this problem and is buggy.

Reply:
:+1: That is a problem. Can you create an bug issue for QueryApi?
If fixing this needs more effort, we can discuss tomorrow how we want to proceed with this PR.

Reply:
Created a bug report for the query API here: https://github.com/camunda-cloud/zeebe/issues/8269

Reply:
Bug is fixed, branch rebased and this change is removed with the lastest commit.